### PR TITLE
Add cache documentation and contributing guide

### PR DESCRIPTION
**Pull request by yiranwang52**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -210,3 +214,7 @@ On the other hand, Makisu has some performance tweaks for large images (especial
 
 BuildKit depends on runc/containerd and supports parallel stage executions, whereas Makisu and most other tools execute Dockefile in order.
 However, BuildKit still needs access to /proc to launch nested containers, which is not ideal and may not be doable in some production environments.
+
+## Contributing
+
+Please checkout our [guide](docs/CONTRIBUTING.md). To contact us, please join our [slack channel](uber-container-tools.slack.com).
```
Comment (apourchet) [inline README.md]: check out

File: README.md [hunk 2]
```diff
@@ -141,14 +142,17 @@ With this job spec, a simple `kubectl create -f job.yaml` will start the build.
...
-It uses target registry for layer storage, and needs to connect to a separate key-value store to map lines of a Dockerfile to a tentative layer SHA stored in Docker registry. For example, Redis can be used as a cache key-value store with this [Kubernetes job spec](examples/k8s/redis.yaml).
-Finally, connect Redis as the Makisu layer cache by passing `--redis-cache-addr=redis:6379` argument.
+Makisu supports distributed cache, which can significantly reduce build time, by up to 90% for some of Uber's code repos.
+Makisu caches docker image layers both locally and in docker registry (if --push parameter is provided), and uses a separate key-value store to map lines of a Dockerfile to names of the layers.
+For example, Redis can be setup as a distributed cache key-value store with this [Kubernetes job spec](examples/k8s/redis.yaml).
+Then connect Makisu to redis cache by passing `--redis-cache-addr=redis:6379` argument.
 Cache has a 7 day TTL by default, which can be configured with `--local-cache-ttl=7d` argument.
+For more options on cache, please see [documentation](docs/CACHE.md)
```
Comment (apourchet) [inline README.md]: add `.` at the end.

File: README.md (hunk 1 above)
Comment (evelynl94) [inline README.md]: Separate contacting from contributing.

File: docs/CACHE.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Cache Configuration
+
+Makisu supports distributed cache, which can significantly reduce build time, by up to 90% for some of Uber's code repos.
+Makisu caches docker image layers both locally and in docker registry (if --push parameter is provided).
+It uses a separate key-value store to map lines of a Dockerfile to names of the layers.
+For cache key-value store, Makisu supports 3 choices:
```
Comment (evelynl94) [inline docs/CACHE.md]: Add a new line here?

File: docs/REGISTRY.md [hunk 4]
```diff
@@ -99,3 +99,5 @@ Example GCR config:
     security:
       credsStore: gcr
 ```
+
+Note: for ECR, environment variable SSL_CERT_DIR is required. Makisu image set that to /makisu-internal/certs by default.
```
Comment (evelynl94) [inline docs/REGISTRY.md]: GCR

**Pull request merged**